tests/integration-tests/conftest.py

```diff
@@ -1223,6 +1223,24 @@ def pytest_runtest_makereport(item, call):
             logging.error("Failed when generating config for failed tests: %s", e, exc_info=True)
 
 
+@pytest.fixture(scope="class")
+def serial_execution_by_instance(request, instance):
+    """Enforce serial execution of tests, according to the adopted instance."""
+    if instance in ["c5n.18xlarge", "p4d.24xlarge"]:
+        logging.info("Enforcing serial execution for instance %s", instance)
+        outdir = request.config.getoption("output_dir")
+        lock_file = f"{outdir}/{instance}.lock"
```

Comment:
Is it output_dir a shared dir for all the tests? or for the single region?
Can you add in the comment an example of output_dir ?

I'm asking this because otherwise we should add the region in the lockfile name to be sure to not block other tests in other regions.

Reply:
output_dir is the regional folder for tests.